homeassistant/components/tradfri.py

```diff
@@ -109,17 +110,22 @@ def gateway_discovered(service, info):
 @asyncio.coroutine
 def _setup_gateway(hass, hass_config, host, key, allow_tradfri_groups):
     """Create a gateway."""
-    from pytradfri import cli_api_factory, Gateway, RequestError, retry_timeout
+    from pytradfri import Gateway, RequestError
+    from pytradfri.api.libcoap_api import api_factory
 
     try:
-        api = retry_timeout(cli_api_factory(host, key))
+        api = api_factory(host, key)
     except RequestError:
+        _LOGGER.error("Tradfri setup failed")
```

Comment:
Maybe this logging isn't even necessary? The default message says about the same:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/setup.py#L205

If you want to add a message about a specific reason for failure that would be appropriate.

Reply:
I agree, default error message will be enough.